File: Rust/output.rs
/*
 *  Output functions for the Rust realizations of  π by quadrature.
 *
 *  Copyright © 2013, 2014  Russel Winder
 */

#![crate_name = "output"]

use std::string::String;
use std::rt::default_sched_threads;

pub fn output(banner:String, pi:f64, n:uint, elapse_time:f64) {
    println!("======================== {}", banner);
    println!("\tπ = {:.18f}", pi);
    println!("\titeration count = {}", n);
    println!("\telapse time = {}", elapse_time);
}

pub fn output_n(banner:String, pi:f64, n:uint, elapse_time:f64, number_of_tasks:uint) {
    output(format!("{}, task count: {}", banner, number_of_tasks), pi, n, elapse_time);
    println!("\tnumber of scheduler threads = {}", default_sched_threads());
}

File: Nim/output.nim
# A module providing an output function for all the variants of the "Π by Quadrature" problem.
#
#  Copyright © 2015 Russel Winder

from strutils import format

proc out_s*(name:string, pi:float, n:int, elapseTime:float) =
  echo format(
    "================ $#\n\tπ = $#\n\titeration count = $#\n\telapse time = $#",
    name, pi, n, elapseTime
  )

proc out_p*(name:string, pi:float, n:int, elapseTime:float, taskCount:int) =
  out_s(format("$# ==== task count: $#", name, taskCount), pi, n, elapseTime)
